crates/subspace-farmer/src/plot.rs

```diff
@@ -510,26 +533,22 @@ impl IndexHashToOffsetDB {
             .delete(&max_distance.to_bytes())
             .map_err(io::Error::other)?;
 
-        self.update_max_distance();
+        self.update_key_bounds();
 
         Ok(result)
     }
 
     fn put(&mut self, index_hash: &PieceIndexHash, offset: PieceOffset) -> io::Result<()> {
-        let distance = PieceDistance::xor_distance(index_hash, &self.address);
+        let key = self.get_key(index_hash);
         self.inner
-            .put(&distance.to_bytes(), offset.to_le_bytes())
+            .put(&key.to_bytes(), offset.to_le_bytes())
             .map_err(io::Error::other)?;
 
-        match self.max_distance {
-            Some(old_distance) => {
-                if old_distance < distance {
-                    self.max_distance.replace(distance);
-                }
-            }
-            None => {
-                self.max_distance.replace(distance);
-            }
+        self.key_bounds = match self.key_bounds {
+            Some((old_lower, upper)) if key < old_lower => Some((key, upper)),
+            Some((lower, old_upper)) if key > old_upper => Some((lower, key)),
+            key_bounds @ Some(_) => key_bounds,
```

Comment:
I'd prefer to have one `Some()` branch here, it should improve readability a bit. Though I see nothing wrong with the logic itself.

Reply:
I just like using guards in match, as you:
- see all the cases on one level
- remove nesting
- read list of cases not list of actions

Should I not use them in general?

Reply:
Everything at one level is sometimes beneficial, but here instead of checking one `match`, which is just `Some` or `None` and then have simple `if` with 3 branches. Combining those into 1 level, especially with different names of variables makes you analyze both conditions on every branch of `match`.

This is subjective of course and code is written correctly, so take it as you wish.

Reply:
Okay, I just wanted to see what is the rule of thumb here :)